### PR TITLE
Specify default JSON:API Extras enhancers link fields globally

### LINKED ISSUE #17021: Specify default JSON:API Extras enhancers link fields globally
## Requirements

We want to be able to override the default output of fields in JSON:API automatically, so that all fields output as we want and new fields will also output correctly.

JSON:API has a concept of 'enhancers', which are data formatters. Rather than outputting the default data structure for a field, an enhancer either adds to the data or reformats it. For example, there is a URL formatter for link fields which adds `url` to the output, which contrasts with `uri`. `uri` may be in a Drupal-internal format that is not usable on the front-end; `url` provides the translation from internal format to general format.

We can override individual field output now. However, in the link field example above, we will always want all link fields to include this information. We want to apply this format to link fields globally, without having to specify it field by field. This will also apply the formatting to new fields as they come into the system.

```[tasklist]
### Acceptance criteria
- [x] For a given JSON:API instance, link field enhancers can be applied globally
- [x] New fields added to the system have the default enhancer applied to them automatically
- [x] Other enhancers can still be applied on a per-field basis
```

## Background & implementation details

This ticket grew out of the need to address things like `field_default_link`, visible in this API response: https://prod.cms.va.gov/jsonapi/paragraph/react_widget/5663a0dd-a69c-40b7-b50b-526646e32204?resourceVersion=id%3A16094

Specifically, these sorts of values:
```
      "field_default_link": {
        "uri": "internal:/pension/application/527EZ",
        "title": "Apply for Veterans Pension Benefits",
        "options": [
          
        ]
      },
```

Questions:
For default enhancers generally:
* Is the a field creation/add event that we can key off to apply the enhancer to new fields?
* Is it important that we be able to apply 'none' as an override from the default enhancer? This would have implications for how we can go about doing the default & overrides.

For link fields specifically:
* Is all link data coming out with the same format from the CMS? Is it a field configuration issue? What are the potential side-effects of changing field configuration, if that is a viable option?
* How many different field configurations do we need to take into account for this change?

### PR DESCRIPTION
**Pull request by alexfinnarn**
## Description

Closes #17021 

## Testing done

Manually tested locally and on Tugboat. Added a functional unit test.

## Screenshots

Screenshot 2024-02-28 at 12 07 40 PM

## QA steps

1. Log into the site and head to "/admin/config/services/jsonapi/field_types".
   - [ ] Confirm that the "link" field type has the "URL for link (link field only)" enhancer applied to it.
   - [ ] "Get Absolute Urls" should be unchecked.
1. Navigate to "/jsonapi/paragraph/react_widget/5663a0dd-a69c-40b7-b50b-526646e32204"
   - [ ] Validate that `field_default_link` has a relative `url` with a value similar to the `uri` field's value.
1. Navigate to "/admin/config/services/jsonapi/field_types", check the "Get Absolute Urls" setting, and save the form.
1. Navigate to "/jsonapi/paragraph/react_widget/5663a0dd-a69c-40b7-b50b-526646e32204"
   - [ ] Validate that `field_default_link` has an absolute `url` with a value similar to the `uri` field's value.
1. Edit "/admin/config/services/jsonapi/add/resource_types/paragraph/react_widget" to add the "URL for link (link field only)" enhancer to the "field_default_link" field. Make sure to uncheck "Get Absolute Urls".
1. Navigate to "/jsonapi/paragraph/react_widget/5663a0dd-a69c-40b7-b50b-526646e32204"
   - [ ] Validate that `field_default_link` has a relative `url` with a value similar to the `uri` field's value.

### Definition of Done

- [x] Documentation has been updated, if applicable.
- [x] Tests have been added if necessary.
- [x] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [x] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [x] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [x] `Accelerated Publishing`

### CODE REVIEW

**Review by tjheffner — DISMISSED** [review]
checked:
Screenshot 2024-02-29 at 9 02 54 AM
unchecked:
Screenshot 2024-02-29 at 9 03 09 AM

can still be overridden at a field level per resource type:
Screenshot 2024-02-29 at 10 02 00 AM

File: patches/field-type-enhancer-3025283-3.patch [hunk 1]
```diff
@@ -0,0 +1,464 @@
+diff --git a/.gitignore b/.gitignore
```
[the remark this answers is not in the document]
Comment (alexfinnarn) [reply patches/field-type-enhancer-3025283-3.patch:1]: yes, this is in the d.o. issue MR and patch.

**Review by tjheffner — APPROVED** [review]
LGTM, again 😁

**Review by edmund-dunn — APPROVED** [review]
I have no issues with this! Nie work!

**Pull request merged**